### PR TITLE
Fix inconsistent rounding in VoxelLineIterator::VoxelLineIterator

### LINKED ISSUE #14554: Pointing problems when you stand exactly at (-10.5, ?, ?)
### Minetest version

```true
5.8.0
```

### Irrlicht device

X11

### Operating system and version

Parabola

### CPU model

_No response_

### GPU model

_No response_

### Active renderer

OpenGL 4.5

### Summary

If you stand at a position where the fractional part of the X or Z coordinate is EXACTLY .5 AND the same coordinate is negative, you can't point nearby nodes. Example numbers: -10.5, -4.5, -0.5.

**Significance**:

Normally, this bug is unlikely to occur since the player has to match that bad coordinate EXACTLY. But a player can get at such a coordinate via `set_pos`. If the player is then unable to move, this bug is suddenly much more visible.

I've discovered this by reworking the bed in Repixture, which puts the player at a .5 coordinate (in the middle of the bed). The workaround was to put the player at a slightly different coordinate.

### Steps to reproduce

1. Generate a flat world
2. Do this command: `/teleport -10.5 ~ ~`
3. Try to point the nodes near you (don't walk!)

Expected result: You can point every node.
Actual result: You can't point the nodes close to you, but the ones further away can be pointed.

Note: It doesn't have to be -10.5. It can be any number `X-0.5`, where X is a negative whole number. This also happens with the Z axis. You can also try to set both X and Z. I haven't tested Y.

The pointing failure stops as soon you move the tiniest bit.

### PR DESCRIPTION
**Pull request by Desour**
Fixes #14554.

## To do

This PR is a Ready for Review.
## How to test

* See #14554.
* `minetest --run-unittests --test-module TestVoxelAlgorithms`

### CODE REVIEW

**Review by sfan5 — APPROVED** [review]
Looks good, consider adding an unit test.

**Comment by Desour** [comment]
Done.
It's kinda amusing how this corner case was explicitly not tested:
>// Do not test lines that start or end on the border of
>// two voxels as rounding errors can make the test fail!

File: src/unittest/test_voxelalgorithms.cpp [hunk 1]
```diff
@@ -49,22 +49,20 @@ void TestVoxelAlgorithms::runTests(IGameDef *gamedef)
...
 		lines.emplace_back(-x, -y, -z, x, y, z);
-	}
-	}
-	}
+	}}}
+	for (f32 x = -3.0; x < 3.1; x += 0.5) {
+	for (f32 y = -3.0; y < 3.1; y += 0.5) {
+	for (f32 z = -3.0; z < 3.1; z += 0.5) {
+		lines.emplace_back(-x, -y, -z, x, y, z);
+	}}}
```
Comment (sfan5) [inline src/unittest/test_voxelalgorithms.cpp:62]: btw
```suggestion
	for (f32 x = -3.0f; x < 3.1f; x += 0.5f)
	for (f32 y = -3.0f; y < 3.1f; y += 0.5f)
	for (f32 z = -3.0f; z < 3.1f; z += 0.5f) {
		lines.emplace_back(-x, -y, -z, x, y, z);
	}
```
